designs/extensible-cli-and-scaffolding-plugins-phase-2.md

```diff
@@ -0,0 +1,203 @@
+# Extensible CLI and Scaffolding Plugins - Phase 2
+
+## Overview
+
+* Plugin Phase 1.5 was desgined and implemented to achieve chaining of plugins. The main goal of Phase 2 plugins is Kubebuilder discovering and using external plugins, also referred as out-of-tree plugins. So Phase 2 achieves both chaining of plugins and discovery of external plugins/source code which is not intented to be shipped with the binaries. For e.g., Kubebuilder being able to discover the helm plugin and use that without having to build the plugin with the binary. These external plugins can be implemented in any language.
+
+### Related issues and PRs
+
+* [Feature Request: Plugins Phase 2](https://github.com/kubernetes-sigs/kubebuilder/issues/1378)
+* [Extensible CLI and Scaffolding Plugins - Phase 1.5](https://github.com/kubernetes-sigs/kubebuilder/blob/master/designs/extensible-cli-and-scaffolding-plugins-phase-1-5.md)
+* [Phase 1.5 Implementation PR](https://github.com/kubernetes-sigs/kubebuilder/pull/2060)
+* [Plugin Resolution Enhancement Proposal](https://github.com/kubernetes-sigs/kubebuilder/pull/1942)
+
+### Use Cases
+
+* As a CLI user, I would like to be able to provide external source code of my plugins and its boilerplates for the CLI to be able to use them by extending the subcommands provided by Kubebuilder, which will not have the plugin binaries be shipped by default.
```

Comment:
```suggestion
* I am an operator author, I would like to be able to provide external plugins path for the CLI to perform the scaffolds, so that I could take advantage of external initiatives which are implemented using Kubebuilder as a lib and following its standards but are not shipped with its CLI binaries. 
```

Reply:
Agree with @jmrodri. Aside: this goal is not "as an operator author", it's "as a plugin developer".

Reply:
That is an option to solve as well. So, you folks are thinking of a manifest that describes the command that would be called in each scenario, the flags and etc. if it is the preferable option then, I think that the others need to be an alternative section and we need here:

- examples of this file
- how will be the contract 
- how the CLI will use it and etc .. 

I mean more detailed info about its design.